.github/workflows/codeql.yml

```
@@ -0,0 +1,60 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master" ]
+    paths-ignore:
+      - '**/*.md'
+      - '**/*.txt'
+      - '**/*.yaml'
+      - '**/*.json'
+      - '**/node_modules'
+      - '**/*.test.js'
+      - '**/*.test.ts'
+  pull_request:
+    branches: [ "master" ]
+    paths-ignore:
+      - '**/*.md'
+      - '**/*.txt'
+      - '**/*.yaml'
+      - '**/*.json'
+      - '**/node_modules'
+      - '**/*.test.js'
+      - '**/*.test.ts'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: 'ubuntu-latest'
+    timeout-minutes: 360
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'javascript-typescript']
```

Prefer to avoid square brackets syntax:

```suggestion
        language:
          - 'javascript-typescript'
```